apigee_edge.module

```diff
@@ -1317,6 +1317,13 @@ function apigee_edge_user_presave(UserInterface $account) {
     }
     $developer = $result->getDeveloper();
     $developer->save();
+
+    // Check if apigee_m10n_add_credit module enabled and org is ApigeeX
+    // if yes call 'default_billingType' hook.
+    if (\Drupal::moduleHandler()->moduleExists('apigee_m10n_add_credit') && \Drupal::service('apigee_m10n.monetization')->isOrganizationApigeeXorHybrid()) {
```

Comment:
Why the Apigee Edge module needs to know about the MINT module or ANY other module that depends on it? It does not. This is a missinterpetaition of the dependency grapth between components.

If you see a need to react on an event when a user is converted to a developer create a _generic_ event to _this_ module, a new public API that can be leveraged by ANY dependent. Do not add hidden/secret checks.

Reply:
This is how we solved things in Drupal 7 and before but that is not necessary anymore in Drupal 8+.

Actually, I realized there is no need for a new event and there is a good reason why we haven't created one before. User -> Developer conversion is handled by `apigee_edge.converter.user_developer` service and in Drupal 8+ you can use a good-old Symfony trick called "service decorator" to exend or alter the public behavior of that service, in other words the created developer from a user. I would propose to replace the current solution with that in the apigee_m10n_add_credit module, no change is needed here related to this.

Reply:
Thanks @mxr576 , thats a great suggestion , I tried and worked as expected.I am going to close this PR and implement the change  in apigee_m10n_add_credit module.
Thanks you so much for your contribution.

Reply:
thank you, happy to help anytime :)